operator/pkg/controllers/addons/addons.go

```diff
@@ -50,7 +49,7 @@ func (c *Controller) Reconcile(ctx context.Context, controlPlane *v1alpha1.Contr
 		return err
 	}
 	// reconcile addons to the guest cluster
-	for _, resource := range []reconciler.Interface{
+	for _, resource := range []v1alpha1.ReconcileFinalize{
```

Comment:
why do you have to change to this interface defined in controlplane api for addons ?

Reply:
I need and interface which has both Reconcile and Finalize for the tests to succeed.